src/hotspot/share/opto/superword.cpp

```diff
@@ -38,146 +38,51 @@
-  _data_entry(arena(), 8,  0, nullptr),                     // nodes with all inputs from outside
-  _mem_slice_head(arena(), 8,  0, nullptr),                 // memory slice heads
-  _mem_slice_tail(arena(), 8,  0, nullptr),                 // memory slice tails
-  _node_info(arena(), 8,  0, SWNodeInfo::initial),          // info needed per node
-  _clone_map(phase->C->clone_map()),                        // map of nodes created in cloning
+SuperWord::SuperWord(Arena* arena, const VLoop &vloop) :
+  _vloop(vloop),
+  _arena(arena),
+  _packset(arena, 8,  0, nullptr),                          // packs for the current block
+  _bb_idx(arena, vloop.estimated_node_count(), 0, 0),       // node idx to index in bb
+  _block(arena, 8,  0, nullptr),                            // nodes in current block
+  _data_entry(arena, 8,  0, nullptr),                       // nodes with all inputs from outside
+  _mem_slice_head(arena, 8,  0, nullptr),                   // memory slice heads
+  _mem_slice_tail(arena, 8,  0, nullptr),                   // memory slice tails
+  _node_info(arena, 8,  0, SWNodeInfo::initial),            // info needed per node
+  _clone_map(phase()->C->clone_map()),                      // map of nodes created in cloning
   _align_to_ref(nullptr),                                   // memory reference to align vectors to
-  _dg(_arena),                                              // dependence graph
-  _nlist(arena(), 8, 0, nullptr),                           // scratch list of nodes
-  _lpt(nullptr),                                            // loop tree node
-  _lp(nullptr),                                             // CountedLoopNode
-  _loop_reductions(arena()),                                // reduction nodes in the current loop
-  _bb(nullptr),                                             // basic block
-  _iv(nullptr),                                             // induction var
+  _dg(arena),                                               // dependence graph
+  _nlist(arena, 8, 0, nullptr),                             // scratch list of nodes
+  _loop_reductions(arena),                                  // reduction nodes in the current loop
   _race_possible(false),                                    // cases where SDMU is true
-  _early_return(true),                                      // analysis evaluations routine
-  _do_vector_loop(phase->C->do_vector_loop()),              // whether to do vectorization/simd style
+  _do_vector_loop(phase()->C->do_vector_loop()),            // whether to do vectorization/simd style
   _num_work_vecs(0),                                        // amount of vector work we have
   _num_reductions(0)                                        // amount of reduction work we have
 {
 }
 
-//------------------------------transform_loop---------------------------
-bool SuperWord::transform_loop(IdealLoopTree* lpt, bool do_optimization) {
```

Comment:
Note: code moved to `VLoop::check_preconditions_helper`

Reply:
Note: all the `do_optimization` parts are not part of preconditions, and hence they are moved to `transform_loop`